registry.js

```diff
@@ -228,16 +228,16 @@ export class ClipboardEntry {
 
             let file = Gio.file_new_for_path(filename);
 
-            const contentType = await new Promise((resolve, reject) => file.query_info_async(null, (obj, res) => {
+            const contentType = await file.query_info_async('*', FileQueryInfoFlags.NONE, GLib.PRIORITY_DEFAULT, null, (obj, res) => {
                 try {
                     const fileInfo = obj.query_info_finish(res);
-                    resolve(fileInfo.get_content_type());
+                    return fileInfo.get_content_type();
                 } catch (e) {
-                    reject(e);
+                    console.error(e);
                 }
-            }));
+            });
 
-            if (!contentType.startsWith('image/') && !contentType.startsWith('text/')) {
+            if (contentType && !contentType.startsWith('image/') && !contentType.startsWith('text/')) {
```

Comment:
In what situations is there no `contentType`?
Also, if there's no content type, will it treat the file as an image? I'm not sure that's what we want, right?

Reply:
I see, I was just add it so that the operations like `startsWith` wouldn't throw an error. About the behavior, it would act the same as the old code, but I think that's not what we want, might open another PR to optimize that later